xsl/common/autoidx-kosek.xsl

```diff
@@ -150,4 +150,157 @@
+      
+      <xsl:for-each select="$l10n.xml">
+        <xsl:variable name="l10n.normalize.sort" select="document(key('l10n-lang', $lang)/@href)/l:l10n/l:gentext[starts-with(@key, 'normalize.sort')]"/>
+        
+        <xsl:choose>
+          <xsl:when test="count($local.l10n.normalize.sort) &gt; 0">
+            <xsl:copy-of select="$local.l10n.normalize.sort"/>
+          </xsl:when>
+          <xsl:when test="count($l10n.normalize.sort) &gt; 0">
+            <xsl:copy-of select="$l10n.normalize.sort"/>
+          </xsl:when>
+          <xsl:otherwise>
+            <xsl:message>
+              <xsl:text>No "</xsl:text>
+              <xsl:value-of select="$lang"/>
+              <xsl:text>" normalizing translate strings exists</xsl:text>
+              <xsl:choose>
+                <xsl:when test="$lang = 'en'">
+                  <xsl:text>.</xsl:text>
+                </xsl:when>
+                <xsl:otherwise>
+                  <xsl:text>; using "en".</xsl:text>
+                </xsl:otherwise>
+              </xsl:choose>
+            </xsl:message>
+            <xsl:copy-of select="document(key('l10n-lang', 'en')/@href)/l:l10n/l:gentext[starts-with(@key, 'normalize.sort.input')]"/>
+          </xsl:otherwise>
+        </xsl:choose>
+      </xsl:for-each>
+    </xsl:variable>
+    
+    <xsl:variable name="normalize.sort.input">
+      <xsl:value-of select="exslt:node-set($normalize.sort)/l:gentext[@key = 'normalize.sort.input']/@text"/>
+    </xsl:variable>
+    <xsl:variable name="normalize.sort.output">
+      <xsl:value-of select="exslt:node-set($normalize.sort)/l:gentext[@key = 'normalize.sort.output']/@text"/>
+    </xsl:variable>
+    
+    <xsl:variable name="term-index">
+      <xsl:call-template name="calculate-term-index">
+        <xsl:with-param name="term" select="$term"/>
+        <xsl:with-param name="term-index" select="' '"/>
+        <xsl:with-param name="letters" select="$letters"/>
+        <xsl:with-param name="normalize.sort.input" select="$normalize.sort.input"/>
+        <xsl:with-param name="normalize.sort.output" select="$normalize.sort.output"/>
+      </xsl:call-template>
+    </xsl:variable>
+    
+    <func:result select="normalize-space($term-index)"/>
+  </func:function>
+  
+  <xsl:template name="calculate-term-index">
+    <xsl:param name="term"/>
+    <xsl:param name="term-index"/>
+    <xsl:param name="letters"/>
+    <xsl:param name="normalize.sort.input"/>
+    <xsl:param name="normalize.sort.output"/>
+    
+    <xsl:variable name="firstletterinterm" select="substring($term, 1, 1)"/>
+    <xsl:variable name="firstletterindex">
+      <xsl:choose>
+        <xsl:when test="$letters/l:l[. = $firstletterinterm]/@i">
+          <xsl:variable name="indexnumber" select="$letters/l:l[. = $firstletterinterm]/@i"/> 
+          <xsl:if test="$indexnumber &lt; 10">
+            <xsl:text>0</xsl:text>
```

Comment:
Why zero is returned if number from lookup table is smaller then 10? There are localization files using numbers <10 for normal characters.